src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcTransport.cs

```diff
@@ -297,7 +298,18 @@ private static bool TryGetDefaultAddress(int pid, out string defaultAddress)
             {
                 try
                 {
-                    defaultAddress = Directory.GetFiles(IpcRootPath, $"dotnet-diagnostic-{pid}-*-socket") // Try best match.
```

Comment:
You're right, good point. Since dotnet-trace never actually required the PID to be visibile in /proc, we could always just share the `/tmp` and use `--process-id` with the NSpid of the process.

I think what will work is - if this path fails (i.e we don't have a valid socket in the `/proc/pid/root/tmp/...nspid...` path) we will fall back to the old behavior. Makes sense?

Reply:
That seems like it would work but be aware that in the above environment if two containers are both running managed apps as lead processes, they will both have a pid of 1. This conflict also exists for tmp mounting though.